src/puyapy_mocks/_ctx_state.py

```diff
@@ -0,0 +1,31 @@
+import attrs
+
+
+@attrs.define
+class ExecutionContextState:
+    args: list[bytes] = attrs.field(factory=list)
+    logs: list[bytes] = attrs.field(factory=list)
+    balances: dict[str, dict[str, int]] = attrs.field(factory=dict)
+
+    def adjust_balance(self, address: str, asset_id: int, amount: int) -> None:
+        if address not in self.balances:
+            self.balances[address] = {}
+        if str(asset_id) not in self.balances[address]:
+            self.balances[address][str(asset_id)] = 0
+        self.balances[address][str(asset_id)] += amount
```

Comment:
wouldn't using asset_id as int instead of str when used as a key in a dict be a simpler option? Given that by default algod and indexer return the field as a 'number'

Reply:
Yep it would.